src/matlab2tikz.m

```diff
@@ -2617,36 +2618,36 @@ case size(cData,2) % specific y-data is given
 function [m2t,posString] = getPositionOfText(m2t, h)
 % makes the tikz position string of a text object
     pos   = get(h, 'Position');
-    npos  = length(pos);
     units = get(h, 'Units');
     xlim  = getOrDefault(m2t.currentHandles.gca, 'XLim',[-Inf +Inf]);
     ylim  = getOrDefault(m2t.currentHandles.gca, 'YLim',[-Inf +Inf]);
     zlim  = getOrDefault(m2t.currentHandles.gca, 'ZLim',[-Inf +Inf]);
     
-    is3D = false;
+    is3D = m2t.axesContainers{end}.is3D;
     
-    switch npos
-        % Check if 3D
-        case 3
-            if pos(3) ~= 0
-                is3D = true;
+    try 
+        cl = class(handle(h));
```

Comment:
Sorry, forgot that `handle()` is not supported. I am looking into it. For the PR do not worry for now.